Library/Homebrew/rubocops/text.rb

```diff
@@ -78,6 +78,21 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
               problem "Do not concatenate paths in string interpolation"
             end
           end
+
+          find_strings(body_node).each do |n|
+            next unless regex_match_group(n, /JAVA_HOME/i)
+
+            next if @formula_name.match?(/^openjdk(@|$)/)
+
+            next if find_every_method_call_by_name(body_node, :depends_on).any? do |dependency|
+              dependency.each_descendant(:str).count.zero? ||
+              regex_match_group(dependency.each_descendant(:str).first, /^openjdk(@|$)/) ||
+              depends_on?(:java)
+            end
```

Comment:
@MikeMcQuaid should this include `adoptopenjdk`?

Reply:
`rubberband` is the only formula blocking this PR. I thought it needed to have `depends_on :java` added so I made a PR (https://github.com/Homebrew/linuxbrew-core/pull/20685). Looking at it, though, it already depends on `adoptopenjdk`. If `adoptopenjdk` fulfills this requirement, I'll add it to this cop, and then this PR should be good to go.

Reply:
`adoptopenjdk` is a cask so can't be depended on. Not sure where you saw this, I'm maybe missing something here?

Reply:
> Not sure where you saw this

https://github.com/Homebrew/linuxbrew-core/blob/master/Formula/rubberband.rb#L25

```ruby
unless OS.mac?
  depends_on "ladspa-sdk"
  depends_on "fftw"
  depends_on "vamp-plugin-sdk"
  depends_on "adoptopenjdk"
end
```

Reply:
Sorry, missed that! I wonder why that isn't just `depends_on "openjdk"`? Maybe it could/should be?

Reply:
I'll give it a shot in the linuxbrew/core PR.